### PR TITLE
introduce a config flag to control a random seed offset

### PR DESCRIPTION
**Pull request by froystig**
... to make it easier to get an idea for seed-sensitive code. Leaving this intentionally undocumented for now.

Usage:
```python
$ python -c 'import jax; print(jax.random.uniform(jax.random.key(17)))'
0.32842708

$ JAX_RANDOM_SEED_OFFSET=0 python -c 'import jax; print(jax.random.uniform(jax.random.key(17)))'
0.32842708

$ JAX_RANDOM_SEED_OFFSET=92 python -c 'import jax; print(jax.random.uniform(jax.random.key(17)))'
0.06752455
```

### CODE REVIEW

**Review by mattjj — COMMENTED** [review]
How about only using this at the process level via an env var (or absl flag), i.e. don't add it to `_trace_context`, or `_GlobalExtraJitContext`, or `_ThreadLocalExtraJitContext`, or set up hooks for updating that trace state, or define a context manager for it?

EDIT: basically like remat_opt_barrier, or eager_pmap, except don't even make a context manager (unless you need it for tests I guess).

**Comment by froystig** [comment]
Yeah, I wanted to write a test...

**Review by mattjj — APPROVED** [review]

**Comment by mattjj** [comment]
SGTM. I'd rather minimize our global state, especially if we only plan to use this at the process level. Tests can be handled by clearing the jit cache in the test setUp method, for example. But it's not a strong preference. Maybe safer to plumb it everywhere and not risk buggy tests.
